tests/functional/Ticket.php

```diff
@@ -560,6 +560,402 @@ public function testUpdateTicketWithActors(array $actors_input, array $expected_
+                    break;
+            }
+        }
+
+        $this->updateItem(
+            \Ticket::class,
+            $ticket_id,
+            [
+                '_actors' => $actors
+            ]
+        );
+
+        $this->checkActors(
+            \Ticket::getById($ticket_id),
+            array_merge($actors['requester'], $actors['observer'], $actors['assign'])
+        );
+
+        // Enable notifications
+        $CFG_GLPI['use_notifications'] = 1;
+        $CFG_GLPI['notifications_mailing'] = 1;
+
+        $this->integer($CFG_GLPI['use_notifications'])->isEqualTo(1);
+        $this->integer($CFG_GLPI['notifications_mailing'])->isEqualTo(1);
+
+        // Get notification id
+        $this->integer(
+            $notif_id = getItemByTypeName(\Notification::class, 'Add Followup', true)
+        )->isGreaterThan(0);
+
+        // Set notifications targets
+        $notification_settings = [
+            'notifications_id' => $notif_id,
+            'itemtype' => \Ticket::class,
+            '_targets' => [
+                '1_3',      //Requester
+                '1_21',     //Observer
+                '1_2'       //Assign
+            ],
+        ];
+
+        $notif_targets = new \NotificationTarget();
+        $notif_targets->updateTargets($notification_settings);
+
+        $targets = getAllDataFromTable(
+            $notif_targets::getTable(),
+            [
+                'SELECT' => [
+                    'id',
+                    'items_id',
+                ],
+                'WHERE' => [
+                    'notifications_id' => $notif_id
+                ],
+                'ORDER' => [
+                    'items_id ASC'
+                ]
+            ]
+        );
+
+        $targets = array_values($targets);
+
+        // Check that notification targets are correctly set
+        $this->array($targets)->hasSize(3);
+        $this->integer($targets[0]['items_id'])->isEqualTo(2);
+        $this->integer($targets[1]['items_id'])->isEqualTo(3);
+        $this->integer($targets[2]['items_id'])->isEqualTo(21);
+
+        // Check that notification queue is empty
+        $notif_queue = new \QueuedNotification();
+        $this->array(getAllDataFromTable($notif_queue->getTable()))->hasSize(0);
+
+
+        // Add ticket followup to trigger notification
+        $this->createItem(
+            \ITILFollowup::class,
+            [
+                'itemtype' => 'Ticket',
+                'items_id' => $ticket_id,
+                'content'    => 'test',
+                'users_id'   => $user_db_id,
+            ]
+        );
+
+        // Check that there are 3 ticket actors whose notifications are activated
+        $this->array(getAllDataFromTable(
+            \Ticket_User::getTable(),
+            [
+                'SELECT' => '*',
+                'WHERE' => [
+                    'tickets_id' => $ticket_id,
+                    'use_notification' => 1
+                ]
+            ]
+        ))->hasSize(3);
```

Comment:
Counting is not precise enough, you can compare tables with `isEqualTo` (to be completed):
```suggestion
        // Check that there are 3 ticket actors whose notifications are activated
        $this->array(getAllDataFromTable(
            \Ticket_User::getTable(),
            [
                'SELECT' => '*',
                'WHERE' => [
                    'tickets_id' => $ticket_id,
                    'use_notification' => 1
                ]
            ]
        ))->isEqualTo($expected);
```

Reply:
This code has been deleted because another block of code does almost the same thing right after it